pkg/controller/pcideviceclaim/pcideviceclaim_controller.go

```diff
@@ -331,49 +286,144 @@ func (h Handler) rebindAfterReboot() error {
 
 func (h Handler) reconcilePCIDeviceClaims(name string, pdc *v1beta1.PCIDeviceClaim) (*v1beta1.PCIDeviceClaim, error) {
 
-	if pdc == nil || pdc.DeletionTimestamp != nil {
+	if pdc == nil || pdc.DeletionTimestamp != nil || (pdc.Spec.NodeName != h.nodeName) {
 		return pdc, nil
 	}
 
-	if pdc.Spec.NodeName == h.nodeName && !pdc.Status.PassthroughEnabled {
-		newPdc, err := h.attemptToEnablePassthrough(pdc)
-		return newPdc, err
+	// Get the PCIDevice object for the PCIDeviceClaim
+	pd, err := h.getPCIDeviceForClaim(pdc)
+	if pd == nil {
+		return pdc, err
+	}
```

Comment:
```suggestion
	if err != nil {
		return pdc, fmt.Errorf("error enabling device passthrough: %v", err)
	}
```
check and handle error while attempting passthrough